src/proto/h1/dispatch.rs

```diff
@@ -180,12 +186,31 @@ where
         task::yield_now(cx).map(|never| match never {})
     }
 
+    fn check_header_read_timeout(&mut self, cx: &mut task::Context<'_>) -> crate::Result<()> {
+        if let Some(header_read_timeout) = self.header_read_timeout {
+            if let Some(read_timeout_fut) = &mut self.header_read_timeout_fut {
+                if Pin::new(read_timeout_fut).poll(cx).is_ready() {
+                    warn!("read header from client timeout");
+                    return Err(crate::Error::new_header_timeout());
```

Comment:
This line returns an error when trying to compile with just the `http1` and `client` features. You can check it by running `cargo check --no-default-features --features client,http1`